call-home/src/collector/report_models.rs

```diff
@@ -0,0 +1,189 @@
+            Some(volumes) => {
+                let replicas_size_vector = get_replicas_size_vector(volumes.entries);
+                replicas.count_per_volume_percentiles =
+                    Percentiles::new(replicas_size_vector.clone());
+            }
+            None => {}
+        };
+        replicas.count = replica_count as u64;
+        replicas
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Versions {
+    control_plane_version: String,
+}
+impl Versions {
+    pub(crate) fn new(control_plane_version: String) -> Self {
+        Self {
+            control_plane_version,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Percentiles {
+    #[serde(rename = "50%")]
+    pub percentile_50: u64,
+    #[serde(rename = "75%")]
+    pub percentile_75: u64,
+    #[serde(rename = "90%")]
+    pub percentile_90: u64,
+}
+
+impl Percentiles {
+    /// Receives  a Vec<u64> and returns Percentiles
+    pub(crate) fn new(values: Vec<u64>) -> Self {
+        Self {
+            percentile_50: get_percentile(values.clone(), 50),
+            percentile_75: get_percentile(values.clone(), 75),
+            percentile_90: get_percentile(values, 90),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Report {
+    pub k8s_cluster_id: String,
+    pub k8s_node_count: u8,
+    pub product_name: String,
+    pub product_version: String,
+    pub deploy_namespace: String,
+    pub storage_node_count: u8,
+    pub pools: Pools,
+    pub volumes: Volumes,
+    pub replicas: Replicas,
+    pub versions: Versions,
+}
+
+/// Get maximum value from a vector
+fn get_max_value(values: Vec<u64>) -> u64 {
+    values.into_iter().max().unwrap_or_default()
+}
+/// Get minimum value from a vector
+fn get_min_value(values: Vec<u64>) -> u64 {
+    values.into_iter().min().unwrap_or_default()
+}
+/// Get mean of all values from a vector
+fn get_mean_value(values: Vec<u64>) -> u64 {
+    let mut sum = 0.0;
+    for value in values.iter() {
+        sum += *value as f64 / (values.len() as f64);
+    }
+    sum as u64
+}
+
+/// Get percentile value from a vector
+fn get_percentile(mut values: Vec<u64>, percentile: usize) -> u64 {
+    if values.len() > 0 {
+        values.sort();
+        let index_as_f64 = (percentile as f64) * (values.len() - 1) as f64 / 100.0;
+        let index = (percentile * (values.len() - 1)) / 100;
+
+        if index_as_f64 - index as f64 > 0.0 {
+            (values[index] as f64
+                + (index_as_f64 - index as f64) * (values[index + 1] - values[index]) as f64)
+                as u64
+        } else {
+            values[index]
+        }
+    } else {
+        0
+    }
+}
+
+/// Gets a vector containing volume sizes from Vec<VolumeStats>
+fn get_volumes_size_vector(volumes: Vec<Volume>) -> Vec<u64> {
+    let mut volume_size_vector = Vec::new();
```

Comment:
```suggestion
    let mut volume_size_vector = Vec::with_capacity(volumes.len());
```

Can do this for the other ones below as well.